File: bailoteca-backend/src/main/java/com/bailoteca/dto/HorarioClaseRequest.java
package com.bailoteca.dto;

import lombok.Data;

import java.time.LocalTime;

/**
 * Clase que representa una solicitud para crear o actualizar un horario de clase.
 * Contiene información sobre el día de la semana, hora de inicio y hora de fin.
 * 
 * @author Tony Aragón
 * @version 1.0
 * @since 1.0
 */
@Data
public class HorarioClaseRequest {
    private Long id;
    private String diaSemana;
    private LocalTime horaInicio;
    private LocalTime horaFin;
}

File: bailoteca-frontend/app/src/main/java/com/example/bailotecaapp/model/dto/HorarioClaseRequest.kt
package com.example.bailotecaapp.model.dto

import kotlinx.serialization.Serializable

/**
 * DTO que representa un horario asociado a una clase. Utilizado tanto para crear
 * como para editar horarios sin duplicarlos. El campo `id` es opcional y solo se usa
 * al editar horarios existentes.
 */
@Serializable
data class HorarioClaseRequest(
    val id: Long? = null,
    val diaSemana: String,
    val horaInicio: String,
    val horaFin: String
)
